### ISSUE TITLE
OutOfMemory with SQL Server

### ISSUE THREAD

**Issue by vincewu** [confluentinc/kafka-connect-jdbc#375]
When using JDBC Source Connect with SQL Server, it's easily to get to OutOfMemory.

The issue is `poll();` in JdbcSourceTask, when reading large data, the Task consumes the first batchMaxRows, and in the 2nd loop, on executing 'querier.maybeStartQuery(cachedConnectionProvider.getValidConnection())', sql server's jdbc driver will detach the first statement before running 'select 1', in which it will fetch all the data into memory (TDSCommand.detach()), where ooo happens.

Can you move the the getValidConnection into maybeStartQuery?
```  
public void maybeStartQuery() throws SQLException {
    if (resultSet == null) {
      stmt = getOrCreatePreparedStatement(cachedConnectionProvider.getValidConnection());
      resultSet = executeQuery();
      schema = DataConverter.convertSchema(name, resultSet.getMetaData(), mapNumerics);
    }
  }
```

**Comment by gzeureka** [comment]
try add "selectMethod" to SQL Server jdbc connection url.

If this property is set to "cursor," a database cursor is created for each query created on the connection for TYPE_FORWARD_ONLY and CONCUR_READ_ONLY cursors. This property is typically required only if the application generates large result sets that cannot be fully contained in client memory. When this property is set to "cursor," only a limited number of result set rows are retained in client memory. The default behavior is that all result set rows are retained in client memory. This behavior provides the fastest performance when the application is processing all rows.

https://docs.microsoft.com/en-us/sql/connect/jdbc/setting-the-connection-properties

**Comment by vincewu** [comment]
Adding selectMethod=cursor can work around the problem.
